### ISSUE TITLE
quarto incorrectly moves caption location in latex table environment to below Table

### ISSUE THREAD

**Issue by kendonB** [quarto-dev/quarto-cli#9122]
### Bug description

If I include a LaTeX table in my quarto document in LaTeX code, the `caption` object gets moved below the table. 

### Steps to reproduce

````
---
format: pdf
tbl-cap-location: top
keep-tex: true
---
  

\begin{table}
\label{tab-test}

\caption{Test caption.}
\begin{tabular}{lr}
\hline
1 & 2 \\
3 & 4
\hline
\end{tabular}

\end{table}
````

### Expected behavior

Should generate a .tex containing:

```
\begin{table}

\caption{\label{tab-test}Test caption.}

\centering{

\begin{tabular}{lr}
\hline
1 & 2 \\
3 & 4
\hline
\end{tabular}

}

\end{table}%
```

### Actual behavior

This generates a .tex containing:

```
\begin{table}

\centering{

\begin{tabular}{lr}
\hline
1 & 2 \\
3 & 4
\hline
\end{tabular}

}

\caption{\label{tab-test}Test caption.}

\end{table}%
```

### Your environment

quarto 1.4.549
Windows 11

### Quarto check output

```bash
1.4.549
(base) PS C:\Users\KennyBell\Dropbox\projects\flood> quarto check
Quarto 1.4.549
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.4.549
      Path: C:\Users\KennyBell\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (external install)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Using: TinyTex
      Path: C:\Users\KennyBell\AppData\Roaming\TinyTeX\bin\windows\
      Version: 2023

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.10.9 (Conda)
      Path: C:/Users/KennyBell/anaconda3/python.exe
      Jupyter: 5.2.0
      Kernels: python3

[>] Checking Jupyter engine render....OK

[>] Checking R installation...........OK
      Version: 4.3.2
      Path: C:/PROGRA~1/R/R-43~1.2
      LibPaths:
        - C:/Users/KennyBell/AppData/Local/R/win-library/4.3
        - C:/Program Files/R/R-4.3.2/library
      knitr: 1.45
      rmarkdown: 2.25

[>] Checking Knitr engine render......OK
```

**Comment by kendonB** [comment]
A workaround seems to be using a different label style. For example, using `table:test` instead of `tab-test`.

**Comment by cscheid** [comment]
This is a duplicate of #8841, `fixed` on main and available on 1.5 prerelease builds.
